includes/class-wc-stripe-helper.php

```diff
@@ -560,6 +560,31 @@ public static function get_dynamic_statement_descriptor( $statement_descriptor =
 		return $statement_descriptor;
 	}
 
+	/**
+	 * Gets the dynamic bank statement descriptor suffix.
+	 *
+	 * Stripe will automatically append this suffix to the merchant account's bank statement prefix.
+	 *
+	 * @param WC_Order $order The order to generate the suffix for.
+	 * @return string The statement descriptor suffix ("#{order-number}").
+	 */
+	public static function get_dynamic_statement_descriptor_suffix( $order ) {
+		$prefix = WC_Stripe::get_instance()->account->get_card_statement_prefix();
+		$suffix = '';
+
+		if ( method_exists( $order, 'get_order_number' ) && ! empty( $order->get_order_number() ) ) {
+			$suffix = '#' . $order->get_order_number();
+
+			// Stripe requires at least 1 latin (alphabet) character in the suffix so we add the first character of the prefix before the order number.
+			if ( 0 === preg_match( '/[a-zA-Z]/', $suffix ) ) {
+				$suffix = substr( $prefix, 0, 1 ) . ' ' . $suffix;
```

Comment:
I just wanted to add some additional details for the reviewer. Stripe requires the `statement_descriptor_suffix` to have at least 1 latin character in it - ie an letter from the alphabet. 

Normally the `get_order_number()` wouldn't include a letter as it would just be "#1245".  Some folks do customise the order number to include a letter so I've made this conditional based on that. 

If the order number doesn't include a letter. I just took the prefix from the Stripe account settings and added the first letter.

I considered an "O" for order. I considered the customers first name initial. I didn't like either of those two options though. Feedback is very welcome :)